raiden/tests/integration/test_endpointregistry.py

```diff
@@ -41,3 +44,43 @@ def test_endpointregistry(private_keys, blockchain_services):
 
     with pytest.raises(KeyError):
         contract_discovery.get(unregistered_address)
+
+
+@pytest.mark.parametrize('number_of_nodes', [3])
+def test_endpointregistry_gas(private_keys, blockchain_services):
+    chain = blockchain_services.blockchain_services[0]
+
+    endpointregistry_address = chain.deploy_contract(
+        'EndpointRegistry',
+        get_contract_path('EndpointRegistry.sol'),
+    )
+    registry_address_hex = address_encoder(endpointregistry_address)
+
+    def get_contract_calls(block, later_than=0):
+        return [tx for tx in block['transactions']
+                if tx['to'] == registry_address_hex and int(block['number'], 16) > later_than]
+
+    prev_block = 0
+    for i in range(len(private_keys)):
+        chain = blockchain_services.blockchain_services[i]
+        discovery_proxy = chain.discovery(endpointregistry_address)
+
+        my_address = privatekey_to_address(private_keys[i])
+        contract_discovery = ContractDiscovery(my_address, discovery_proxy)
+        contract_discovery.register(my_address, '127.0.0.1', 44444)
+
+        block = chain.client.find_block(lambda block: get_contract_calls(block, prev_block))
```

Comment:
This is failing with the tester implementation, it's better to expose the `find_block` call to the chain proxy, so that both implementation can pass.